archive/oauth-and-openid.md

```diff
@@ -57,17 +57,88 @@ With this process
 * Client doesn't need to request with Resource Owner's credential.
 * Resource Server doesn't need to authenticate user always.
 
+# Obtaining Authorization
 
-## OAuth 2.1 does not recommend Implicit Grant
+## Authorization Code Grant
 
-OAuth 2.0 supports [Implicit Grant](https://datatracker.ietf.org/doc/html/rfc6749#section-1.3.2), however,
-OAuth 2.1 doesn't recommend it but recommend trying authorization request with
-[PKCE](https://datatracker.ietf.org/doc/html/draft-ietf-oauth-v2-1-01#oauth-2-0-differences). Moreoever, 
-there are other considerations in OAuth flow.  OAuth 2.1 already expired, however, those approached are problem 
-to think about when providing OAuth flow.
+This grant type is used to obtain access token and refresh token. It's important to create redirection-based flow
+with confidential client.
 
+* Resource owner: Target to grant to access resource to the client
+* User-agent: User agent is a browser or mobile application via which the resource owner communicates to the
+  authorization server.
+* Client: Client is the application code that wants to access the resources of the user on the resource server.
+  Client can be a web browser, native client or web server. If the client is on the server, it's considered a
+  confidential client. If not it's public client.
+
+
+     +----------+
+     | Resource |
+     |   Owner  |
+     |          |
+     +----------+
+          ^
+          |
+         (B)
+     +----|-----+          Client Identifier      +---------------+
+     |         -+----(A)-- & Redirection URI ---->|               |
+     |  User-   |                                 | Authorization |
+     |  Agent  -+----(B)-- User authenticates --->|     Server    |
+     |          |                                 |               |
+     |         -+----(C)-- Authorization Code ---<|               |
+     +-|----|---+                                 +---------------+
+       |    |                                         ^      v
+      (A)  (C)                                        |      |
+       |    |                                         |      |
+       ^    v                                         |      |
+     +---------+                                      |      |
+     |         |>---(D)-- Authorization Code ---------'      |
+     |  Client |          & Redirection URI                  |
+     |         |                                             |
+     |         |<---(E)----- Access Token -------------------'
+     +---------+       (w/ Optional Refresh Token)
```

Comment:
This will be shown as broken diagram in github. Please use code block.

![image](https://github.com/user-attachments/assets/5fdf57c0-bfaf-4931-86ed-8e2bf6c97df4)